### PR TITLE
Refactor/node names

### PR DESCRIPTION
**Pull request by Yadunund**
## Refactor

What started out as a simple fix for appropriately renaming the ROS 2 nodes and executable (#2) turned out to be a complete refactor of `rmf_schedule_visualizer`.

Summary of changes:
* `src/VisualizerData.hpp` renamed to `ScheduleDataNode.hpp` and moved into public API while adopting PIMPL pattern
* `src/Server.hpp` renamed to `TrajectoryServer.hpp` and moved into public API while adopting PIMPL pattern
* Exportable `rmf_schedule_visualizer` library of public API
* Single `main.cpp` which publishes RVIZ markers for schedule visualization and starts the websocket server to serve trajectories
* Renamed ROS 2 executable from `rviz2` to `schedule_visualizer`
* Renamed nodes `/viz` and `rviz_node` to `rmf_schedule_visaulizer_data_node` and `rmf_schedule_visualizer_marker_publisher`
* Parse `headless` argument in `visualizer.launch.xml` to determine whether to launch `RViz`

### CODE REVIEW

**Comment by Yadunund** [comment]
@koonpeng could you please test this branch to make sure nothing is broken for the dashboard?
@ddengster could you also test this branch to make sure nothing is broken with the negotiation viz/server?

**Comment by Yadunund** [comment]
This branch is required for testing with `rmf_demos`
https://github.com/open-rmf/rmf_demos/pull/11

**Comment by ddengster** [comment]
> @ddengster could you also test this branch to make sure nothing is broken with the negotiation viz/server?

Doesnt seem like anything's changed on this backend server, should be GTG

File: rmf_schedule_visualizer/include/rmf_schedule_visualizer/ScheduleDataNode.hpp
Comment (gbiggs) [inline rmf_schedule_visualizer/include/rmf_schedule_visualizer/ScheduleDataNode.hpp:85]: I won't refuse an approval over this, but in general the node should decide its name, not the function that constructs it. If the node name needs to be changed remapping can be used.

A node name is a data type name, not a variable name. The node instance name is the variable name and remapping is how you name the variable.

File: rmf_schedule_visualizer/src/main.cpp
Comment (gbiggs) [inline rmf_schedule_visualizer/src/main.cpp:53]: This is huge. Consider moving it to a separate file.

File: rmf_schedule_visualizer/src/main.cpp
Comment (gbiggs) [inline rmf_schedule_visualizer/src/main.cpp:904]: Why 3?

File: rmf_schedule_visualizer/src/rmf_schedule_visualizer/ScheduleDataNode.cpp
Comment (gbiggs) [inline rmf_schedule_visualizer/src/rmf_schedule_visualizer/ScheduleDataNode.cpp:26]: Minor nitpick: Place includes in alphabetical order.

File: rmf_schedule_visualizer/src/rmf_schedule_visualizer/ScheduleDataNode.cpp
Comment (gbiggs) [inline rmf_schedule_visualizer/src/rmf_schedule_visualizer/ScheduleDataNode.cpp:113]: Why is the result of this comparison stored in a variable that's only used in the very next line?

File: rmf_schedule_visualizer/src/rmf_schedule_visualizer/ScheduleDataNode.cpp
Comment (gbiggs) [inline rmf_schedule_visualizer/src/rmf_schedule_visualizer/ScheduleDataNode.cpp:81]: Consider moving some of the `Implementation` setup into that class to shorten this function.

File: rmf_schedule_visualizer/src/rmf_schedule_visualizer/TrajectoryServer.cpp
Comment (gbiggs) [inline rmf_schedule_visualizer/src/rmf_schedule_visualizer/TrajectoryServer.cpp:354]: Consider moving some of this inside the pimpl object to shorten and simplify this function.

**Review by gbiggs — COMMENTED** [review]
In general it's good, but I have a couple of comments I'd like you to look at before I hit approve.

File: rmf_schedule_visualizer/include/rmf_schedule_visualizer/ScheduleDataNode.hpp
Comment (Yadunund) [reply rmf_schedule_visualizer/include/rmf_schedule_visualizer/ScheduleDataNode.hpp]: Thanks for the heads up. Will keep this in mind in the future

File: rmf_schedule_visualizer/src/main.cpp
Comment (Yadunund) [reply rmf_schedule_visualizer/src/main.cpp]: Fixed. This should be 2. Thanks for pointing it out.

**Review by gbiggs — APPROVED** [review]

**Pull request merged**
